examples/src/main/kotlin/io/aexp/bucketlist/examples/prlifetime/ExportRawPrLifetimeData.kt

```diff
@@ -0,0 +1,125 @@
+
+        val configPath = Paths.get(args[0])
+        val client = getBitBucketClient(configPath)
+
+        val projectKey = args[1]
+        val repoSlug = args[2]
+        val outputCsvPath = Paths.get(args[3])
+
+        val startDate = ZonedDateTime.of(LocalDateTime.of(LocalDate.parse(args[4]), LocalTime.MIDNIGHT), ZoneId.of("UTC"))
+        val endDate = ZonedDateTime.of(LocalDateTime.of(LocalDate.parse(args[5]), LocalTime.MIDNIGHT), ZoneId.of("UTC"))
+
+        val prSummaries = getPrSummaries(client, projectKey, repoSlug, startDate, endDate)
+                .toSortedList({ prSummary1, prSummary2 -> prSummary1.pr.createdAt.compareTo(prSummary2.pr.createdAt) })
+                .toBlocking()
+                .first()
+
+        printBetterCsv(outputCsvPath, prSummaries)
+        System.exit(0)
+    }
+
+    /**
+     * Loads PrSummary objects for every PR that was merged within the specified date range
+     *
+     * @return an observable of every PrSummary for the corresponding date range
+     */
+    private fun getPrSummaries(client: BucketListClient, projKey: String, repoSlug: String, startDate: ZonedDateTime,
+                               endDate: ZonedDateTime) : Observable<PrSummary> {
+        return client.getPrs(projKey, repoSlug, PullRequestState.MERGED, Order.OLDEST)
+                .observeOn(Schedulers.io())
+                // flatten pages into one stream of prs
+                .flatMap({ prs -> Observable.from(prs.values) })
+                .filter({ pr ->
+                    // Don't bother getting activity for any PRs that were created after the end date
+                    // Don't bother getting activity for any PRs that are closed, where the last update was before the start date.
+                    pr.createdAt.isBefore(endDate) && !(pr.closed && pr.updatedAt.isBefore(startDate))
+                })
+                .flatMap({ pr ->
+                    // gather all activity for each pr into a PrSummary
+                    client.getPrActivity(projKey, repoSlug, pr.id)
+                            .flatMap({ page -> Observable.from(page.values) })
+                            .toSortedList({ a1, a2 -> a1.createdAt.compareTo(a2.createdAt) })
+                            .map({ list -> PrSummary(pr, list) })
+                })
+    }
+
+    /**
+     * Writes creation date and durations for PrSummaries
+     */
+    private fun printBetterCsv(outputCsvPath: Path, prSummaries: List<PrSummary>) {
+        val outputCsvFile = outputCsvPath.toFile()
+        CSVWriter(FileWriter(outputCsvFile)).use { writer ->
+            val header = ArrayList<String>()
+            header.add("Created At")
+            header.add("Duration Since Start (hours)")
+            header.add("Duration Since Last Push (hours)")
+            writer.writeNext(header.toArray(arrayOfNulls<String>(0)))
+
+            for (prSummary in prSummaries) {
+                val row = ArrayList<String>()
+                val pr = prSummary.pr
```

Comment:
You only use this intermediate thing once, it looks like.  You could get rid of it, if you want.  No strong preference from me.